File: 461/main.py
"""
The Hamming distance between two integers is the number of positions at which the corresponding bits are different.

Given two integers x and y, calculate the Hamming distance.

Note:
0 ? x, y < 231.

Example:

Input: x = 1, y = 4

Output: 2

Explanation:
1   (0 0 0 1)
4   (0 1 0 0)
       ?   ?

The above arrows point to positions where the corresponding bits are different.
"""

class Solution(object):
    def hammingDistance(self, x, y):
        """
        :type x: int
        :type y: int
        :rtype: int
        """
        ans = 0
        while x > 0 or y > 0:
            ans += (x & 1) ^ (y & 1)
            x //= 2
            y //= 2
            pass
        return ans

ans = Solution()
print(ans.hammingDistance(1, 4))

File: 461/main.go
package main

import "fmt"

func main() {
	fmt.Println(hammingDistance(1, 4))
}

func hammingDistance(x int, y int) int {
	var answer int
	for x > 0 || y > 0 {
		answer += (x & 1) ^ (y & 1)
		x >>= 1
		y >>= 1
	}
	return answer
}
